Screenshot maps

aparlato opened the pull request:
Closes https://github.com/stamen/map-compare/issues/36

This PR adds a button to download screenshots of the maps you're comparing.

### Note:

It **only** works for mirror and phone modes. This is because we use `html2canvas` which cannot handle css rules for clip path which we use in swipe mode. There's other packages like `dom-to-image` that should be able to handle clipping, but I ran into other CSS oriented errors there and decided to move forward with `html2canvas` since I don't believe we typically screenshot in swipe mode anyway.

Please let me know if there's a solid use case for screenshotting in swipe mode, and I'm happy to revisit, but after some exploration, this felt like the best option for now.

aparlato commented on package.json:
```diff
@@ -28,8 +28,9 @@
     "@beyonk/svelte-mapbox": "^8.1.4",
     "@fortawesome/free-solid-svg-icons": "^6.1.1",
     "@googlemaps/js-api-loader": "^1.13.10",
-    "@mapbox/mapbox-gl-geocoder": "^4.7.4",
```
This was redundant it seems because of `@beyonk/svelte-mapbox`. Let me know if I've misunderstood why this is here though!

ebrelsford commented on src/components/MapControls.svelte:
```diff
@@ -51,6 +55,22 @@
+        return el.className.includes('map-label');
+      }
+      return false;
+    };
```
Removing them for now is great.

ebrelsford reviewed (approved):
This looks great to me.

I'll defer to @kelsey-taylor here but I think only screenshots in phone or mirror is fine.

I'm approving the change but I have a few thoughts about potential improvements, up to you if we do them here or in separate issues:
 * How hard would it be to just copy to your clipboard rather than download a file? Clipboard would be my preference, but I could be an outlier.
 * How hard would it be to clip to the phones when in phone mode? There's a lot of whitespace as-is (depending on your screen):
     ![mapbox-dark-mapbox-light (2)](https://user-images.githubusercontent.com/375002/166507870-375d08aa-c7c4-4f96-99c7-2d24648f5704.png)
    My preference would be to have no margin around the "phones", if relatively easy.

aparlato commented:
@ebrelsford good call! Changed to copy to clipboard and adjusted the phones to be closer together and only copy the necessary space.

![image](https://user-images.githubusercontent.com/15698320/166520028-8e187286-a889-40ae-b99a-a789def5dcda.png)
![image](https://user-images.githubusercontent.com/15698320/166520047-56d48b07-2859-40a9-83c8-604a83f43d1a.png)

ebrelsford commented:
Looks great! What do you think @kelsey-taylor?

kelsey-taylor reviewed (approved):
sorry for the delay here! all of this looks/sounds great to me. 

i'm agnostic about copy to clipboard vs. download - just one is totally fine for now but i suppose i wouldn't mind both someday 🙂

aparlato commented:
> just one is totally fine for now but i suppose i wouldn't mind both someday 🙂

Sounds good @kelsey-taylor ! In that case if you're ok proceeding with one, let's go ahead with the clipboard and we can add on the download if it's a common use case!
